scripts/zkai_attention_kv_stwo_binary_typed_proof_accounting_gate.py

```
@@ -0,0 +1,685 @@
+    if summary["upstream_stwo_serialization_status"] != UPSTREAM_SERIALIZATION_STATUS:
+        raise BinaryTypedProofAccountingGateError("CLI upstream serialization status drift")
+    if summary["proof_payload_kind"] != PROOF_PAYLOAD_KIND:
+        raise BinaryTypedProofAccountingGateError("CLI proof payload kind drift")
+    rows = summary.get("rows")
+    if not isinstance(rows, list) or len(rows) != len(EXPECTED_ROLES):
+        raise BinaryTypedProofAccountingGateError("CLI row count drift")
+    for row, expected in zip(rows, EXPECTED_ROLES, strict=True):
+        validate_cli_row(row, expected)
+
+
+def validate_cli_row(row: Any, expected: dict[str, Any]) -> None:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `row`

(ANN401)